Monal/Classes/MLPromiseRegistry.m

```diff
@@ -0,0 +1,260 @@
+//
+//  MLPromiseRegistry.m
+//  monalxmpp
+//
+//  Created by Matthew Fennell on 29/09/2024.
+//  Copyright © 2024 monal-im.org. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+#import "HelperTools.h"
+#import "MLPromiseRegistry.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation MLSerializablePromise
+
+-(instancetype) init
+{
+    _uuid = [NSUUID UUID];
+    _isResolved = false;
+    return self;
+}
+
+-(void) resolve:(id _Nullable) arg
+{
+    _resolveArg = arg;
+    _isResolved = true;
+}
+
++(BOOL) supportsSecureCoding
+{
+    return YES;
+}
+
+-(void) encodeWithCoder:(NSCoder*) coder
+{
+    [coder encodeObject:_uuid forKey:@"uuid"];
+    [coder encodeObject:_resolveArg forKey:@"resolveArg"];
+    [coder encodeBool:_isResolved forKey:@"isResolved"];
+}
+
+-(nullable instancetype) initWithCoder:(NSCoder*) coder
+{
+    self = [self init];
+    _uuid = [coder decodeObjectForKey:@"uuid"];
+    _resolveArg = [coder decodeObjectForKey:@"resolveArg"];
+    _isResolved = [coder decodeBoolForKey:@"isResolved"];
+    return self;
+}
+
+@end
+
+@interface MLPromise()
+
+@property(nonatomic, strong) MLSerializablePromise* serializablePromise;
+
+@property(nonatomic, weak) MLPromiseRegistry* promiseRegistry;
+
+@property(nonatomic, strong) PromiseResolverCallback promiseResolverCallback;
+@property(nonatomic, strong) PMKResolver onConsumeCallback;
+
+@property(nonatomic, strong) AnyPromise* promise;
+
+@end
+
+@implementation MLPromise
+
+-(instancetype) initWithCallback:(PromiseResolverCallback) promiseResolverCallback andPromiseRegistry:(MLPromiseRegistry*) promiseRegistry
```

Comment:
I plan to clean up these `init` methods as there is a lot of duplication between them

Reply:
ah sorry!! I used the term `interface` (java jargon), but I meant `protocol` in objc terminology.

Reply:
the hierarchy was: `protocol` `MLPromise`, class implementing this protocol (by inheriting from this protocol): `MLSerializablePromise`. My fault, sorry!!